pkgs/by-name/gd/gdmd/package.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, makeWrapper
+, gdc
+, perl
+}:
+stdenvNoCC.mkDerivation {
+  pname = "gdmd";
+  version = "0.1.0-unstable-2024-05-30";
+
+  src = fetchFromGitHub {
+    owner = "D-Programming-GDC";
+    repo = "gdmd";
+    rev = "dc0ad9f739795f3ce5c69825efcd5d1d586bb013";
+    hash = "sha256-Sw8ExEPDvGqGKcM9VKnOI6MGgXW0tAu51A90Wi4qrRE=";
+  };
+
+  patches = [
+    ./0001-gdc-store-path.diff
+  ];
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ gdc perl ];
+
+  postPatch = ''
+    substituteInPlace dmd-script \
+      --replace-fail gdcStorePath ${gdc}
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 dmd-script $out/bin/gdmd
+    install -Dm644 dmd-script.1 $out/share/man/man1/gdmd.1
+    wrapProgram $out/bin/gdmd \
+      --prefix PATH : ${lib.makeBinPath [ gdc perl ]}
```

Comment:
Would it be possible to avoid `wrapProgram`? It's OK for programs which don't need to run other (user-specified) programs, but e.g. `wrapProgram` will cause `gdmd -run user_program.d` to inherit the variables specified in the `wrapProgram` arguments - ideally, `user_program` would see an unmodified environment, without `perl` etc.

Reply:
Ah you're right, the script's shebang is already patched to point to perl's store path and the path to GDC is set as well.

Reply:
Yep it totally works without the wrapping.